### DISCUSSION TITLE
Timestep when calling the function plume_temporal_evolution in Cocipgrid

### DISCUSSION THREAD

**Discussion by Frank251826** [contrailcirrus/pycontrails#243] [Q&A] [answered]
The timestep here directly takes the parameter "dt_integration". For the first contrail, dt are usually not as long as dt_integration. I am wondering if this is something we can modify a little bit. 

Start at line 1674 to 1684 in cocip_grid:

    sigma_yy_t2, sigma_zz_t2, sigma_yz_t2 = contrail_properties.plume_temporal_evolution(
        width_t1=width_t1,
        depth_t1=depth_t1,
        sigma_yz_t1=sigma_yz_t1,
        dsn_dz_t1=dsn_dz_t1,
        diffuse_h_t1=diffuse_h_t1,
        diffuse_v_t1=diffuse_v_t1,
        seg_ratio=seg_ratio_t12,
        dt=params["dt_integration"],
        max_depth=params["max_depth"],
    )

Refer to the same usage in Cocip, dt is defined as    contrail_2['time']-contrail_1['time'],    which is reasonable.
A lot appreciate if anyone could provide some help.

**Comment by zebengberg** [comment]
Great catch! What if we change `dt=params["dt_integration"]` to `dt=next_contrail["time"] - curr_contrail["time"]`? Would that bring the two implementations into harmony?

**Reply by zebengberg** [reply, marked as answer]
@Frank251826 Please let me know what you think: https://github.com/contrailcirrus/pycontrails/pull/245

**Reply by Frank251826** [reply]
I did exactly the same and it works perfectly, which brings same result as Cocip.
